main.go

```diff
@@ -315,6 +316,48 @@ func setPriority(nice int, class string) error {
 	return nil
 }
 
+func openProfile(c *config.Config, profileName string) (profile *config.Profile, cleanup func(), err error) {
+	cleanup = func() {}
+	done := false
+
+	for attempts := 3; attempts > 0 && !done; attempts-- {
+		profile, err = c.GetProfile(profileName)
+		if err != nil || profile == nil {
+			err = fmt.Errorf("cannot load profile '%s': %w", profileName, err)
+			break
+		}
+
+		done = true
+
+		// Adjust baseDir if needed
+		if len(profile.BaseDir) > 0 {
+			var currentDir string
+			currentDir, err = os.Getwd()
+			if err != nil {
+				err = fmt.Errorf("changing base directory not allowed as current directory is unknown in profile %q: %w", profileName, err)
```

Comment:
Is this a problem to change the current directory to `BaseDir` then leave without changing it back to the (unknown) original?

I'm not sure there are valid cases where it could happen though. The only one I can think of is executing a shell script while downloading it (via a unix pipe). I suppose why not?

Reply:
come to think of it... you still get a current directory
```
/tmp % echo "pwd" | sh
/tmp
```